node/pkg/watchers/solana/client.go

```diff
@@ -347,7 +355,7 @@ func (s *SolanaWatcher) Run(ctx context.Context) error {
 					ContractAddress: contractAddr,
 				})
 
-				if s.wsUrl == nil {
+				if useWs {
```

Comment:
Should be !useWs. I think that's why tests are failing.

Reply:
Fixed